sklearn/gaussian_process/tests/test_gpr.py

```diff
@@ -639,28 +665,37 @@ def test_gpr_consistency_std_cov_non_invertible_kernel():
 
 
 @pytest.mark.parametrize(
-    "params, TypeError, err_msg",
+    "gpr_params, fit_params, TypeError, err_msg",
     [
         (
             {"alpha": np.zeros(100)},
+            {"sample_variance": None},
             ValueError,
             "alpha must be a scalar or an array with same number of entries as y",
         ),
+        (
+            {"alpha": np.zeros(100)},
+            {"sample_variance": np.zeros(100)},
+            ValueError,
+            "sample_variance must be a scalar or an array with same number of entries"
+            " as y",
+        ),
         (
             {
                 "kernel": WhiteKernel(noise_level_bounds=(-np.inf, np.inf)),
                 "n_restarts_optimizer": 2,
             },
+            {"sample_variance": None},
```

Comment:
Let's also test the case where `sample_variance` is not passed at all.

```suggestion
            {},
```

Reply:
ok